pkg/azure/types.go

```diff
@@ -153,6 +153,10 @@ const (
 	// CSIMigrationKubernetesVersion is a constant for the Kubernetes version for which the Shoot's CSI migration will be
 	// performed.
 	CSIMigrationKubernetesVersion = "1.21"
+
+	// ForceK8sVersionUpgrade is an annotation set on the Shoot resource that allows to enforce a k8s version upgrade
+	// even if it would otherwise be forbidden by the admission controller.
+	ForceK8sVersionUpgrade = "shoot.gardener.cloud/force-version-upgrade"
```

Comment:
Just optional nit: I think it is more fitting  to `confirmation.gardener.cloud'`